### ISSUE TITLE
WebGL error with custom shapes

### ISSUE THREAD

**Issue by GolddolphinSKB-Fexcraftian** [processing/p5.js#2435]
#### Nature of issue?

- [X] Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [X] WebGL
- [ ] Other (specify if possible)

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [X] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: v0.5.16
- Web browser and version: Chrome 63.0.3239.84
- Operating System: macOS
- Steps to reproduce this:
Error when calling vertices.
```
Uncaught TypeError: Cannot read property 'location' of undefined
    at p5.RendererGL._drawFillImmediateMode (p5.js:68723)
    at p5.RendererGL.endShape (p5.js:68697)
    at p5.endShape (p5.js:51228)
    at draw (sketch.js:358)
    at p5.redraw (p5.js:50266)
    at p5.<anonymous> (p5.js:44917)
```

Here is my draw code:
````javascript
        background(0);
        noStroke();
        ambientLight(10, 100, 50);
        rotateX(frameCount * 0.02)
        ambientMaterial(250);
        for (var i=0; i<vertices.length; i++) {
            beginShape(LINES);
            vertex(render[i].z2, render[i].y2, render[i].x2);
            vertex(render[i].z3, render[i].y3, render[i].x3);
            vertex(render[i].z0, render[i].y0, render[i].x0);
            vertex(render[i].z1, render[i].y1, render[i].x1);
            endShape(CLOSE);

            beginShape(LINES);
            vertex(render[i].z5, render[i].y5, render[i].x5);
            vertex(render[i].z4, render[i].y4, render[i].x4);
            vertex(render[i].z7, render[i].y7, render[i].x7);
            vertex(render[i].z6, render[i].y6, render[i].x6);
            endShape(CLOSE);

            beginShape(LINES);
            vertex(render[i].z0, render[i].y0, render[i].x0);
            vertex(render[i].z3, render[i].y3, render[i].x3);
            vertex(render[i].z7, render[i].y7, render[i].x7);
            vertex(render[i].z4, render[i].y4, render[i].x4);
            endShape(CLOSE);

            beginShape(LINES);
            vertex(render[i].z2, render[i].y2, render[i].x2);
            vertex(render[i].z1, render[i].y1, render[i].x1);
            vertex(render[i].z5, render[i].y5, render[i].x5);
            vertex(render[i].z6, render[i].y6, render[i].x6);
            endShape(CLOSE);

            beginShape(LINES);
            vertex(render[i].z1, render[i].y1, render[i].x1);
            vertex(render[i].z0, render[i].y0, render[i].x0);
            vertex(render[i].z4, render[i].y4, render[i].x4);
            vertex(render[i].z5, render[i].y5, render[i].x5);
            endShape(CLOSE);

            beginShape(LINES);
            vertex(render[i].z3, render[i].y3, render[i].x3);
            vertex(render[i].z2, render[i].y2, render[i].x2);
            vertex(render[i].z6, render[i].y6, render[i].x6);
            vertex(render[i].z7, render[i].y7, render[i].x7);
            endShape(CLOSE);
        }
    }
}
```
I've also tried using Quads and triangle strips, but they produced the same error.
I'm not entirely sure on where the error is actually from...
Any help will be greatly appreciated.

**Comment by groundrace** [comment]
I've not checked the details but looks like this issues start appearing from 0.5.15 version
and is tight correlated with the usage of WebGL

**Comment by magnifi** [comment]
I think the issue is caused by the vertex color buffer --`aVertexColor`-- not being initialized, unless you call `fill()`.

**Comment by fractalfrenzy** [comment]
Even if you call fill(), the error still appears. Try this code:

```
function setup() {
	createCanvas(100, 100, WEBGL);
}

function draw() {
	fill(0);
	background(200);
	rotateX(frameCount * 0.01);
 	rotateY(frameCount * 0.01);
  	box(20, 20, 20);
}
```

**Comment by Spongman** [comment]
Can you see of this is a dupe of #2262? There's a zip with a build of p5 that _should_ fix this in that thread.

**Comment by mlarghydracept** [comment]
I am going to go ahead and close this because it seems likely that this is a duplicate of #2262 which was closed by #2349. If you try it with the latest changes in the github repo (or try the zip file over at #2262) and the problem is still present just comment here and I will reopen. Thanks!
